core/synchronizer.ts

```diff
@@ -0,0 +1,805 @@
+        this.socialConfig = config
+    }
+
+    async loadNewEvents(fromBlock, toBlock) {
+        return (
+            (await Promise.all([
+                super.loadNewEvents(fromBlock, toBlock),
+                this.unirepSocialContract.queryFilter(
+                    this.unirepSocialFilter,
+                    fromBlock,
+                    toBlock
+                ),
+            ])) as any
+        ).flat()
+    }
+
+    get topicHandlers(): any {
+        const [_UserSignedUp] = this.unirepSocialContract.filters.UserSignedUp()
+            .topics as string[]
+        const [_PostSubmitted] =
+            this.unirepSocialContract.filters.PostSubmitted().topics as string[]
+        const [_CommentSubmitted] =
+            this.unirepSocialContract.filters.CommentSubmitted()
+                .topics as string[]
+        const [_VoteSubmitted] =
+            this.unirepSocialContract.filters.VoteSubmitted().topics as string[]
+        const [_AirdropSubmitted] =
+            this.unirepSocialContract.filters.AirdropSubmitted()
+                .topics as string[]
+        return {
+            ...super.topicHandlers,
+            [_UserSignedUp]: this.socialUserSignedUp.bind(this),
+            [_PostSubmitted]: this.postSubmittedEvent.bind(this),
+            [_CommentSubmitted]: this.commentSubmittedEvent.bind(this),
+            [_VoteSubmitted]: this.voteSubmittedEvent.bind(this),
+            [_AirdropSubmitted]: this.airdropSubmittedEvent.bind(this),
+        }
+    }
+
+    get unirepSocialFilter() {
+        const [_UserSignedUp] = this.unirepSocialContract.filters.UserSignedUp()
+            .topics as string[]
+        const [_PostSubmitted] =
+            this.unirepSocialContract.filters.PostSubmitted().topics as string[]
+        const [_CommentSubmitted] =
+            this.unirepSocialContract.filters.CommentSubmitted()
+                .topics as string[]
+        const [_VoteSubmitted] =
+            this.unirepSocialContract.filters.VoteSubmitted().topics as string[]
+        const [_AirdropSubmitted] =
+            this.unirepSocialContract.filters.AirdropSubmitted()
+                .topics as string[]
+        // Unirep Social events
+        return {
+            address: this.unirepSocialContract.address,
+            topics: [
+                [
+                    _UserSignedUp,
+                    _PostSubmitted,
+                    _CommentSubmitted,
+                    _VoteSubmitted,
+                    _AirdropSubmitted,
+                ],
+            ],
+        }
+    }
+
+    async socialUserSignedUp(event, db) {
+        const _epoch = Number(event.topics[1])
+        const _commitment = BigInt(event.topics[2]).toString()
+        db.create('UserSignUp', {
+            transactionHash: event.transactionHash,
+            commitment: _commitment,
+            epoch: _epoch,
+        })
+    }
+
+    private async verifyAttestationProof(
```

Comment:
If it is verified by `@unirep/core` Synchronizer, we can remove the check here.

Reply:
This is fixed, but it needs unirep/unirep#58 to work properly

Reply:
Moved to #11 so we can merge this